lib/std/os/bits/linux/sparc64.zig

```diff
@@ -0,0 +1,679 @@
+pub const O_TRUNC = 0x400;
+pub const O_APPEND = 0x8;
+pub const O_NONBLOCK = 0x4000;
+pub const O_SYNC = 0x802000;
+pub const O_DSYNC = 0x2000;
+pub const O_RSYNC = O_SYNC;
+pub const O_DIRECTORY = 0x10000;
+pub const O_NOFOLLOW = 0x20000;
+pub const O_CLOEXEC = 0x400000;
+
+pub const O_ASYNC = 0x40;
+pub const O_DIRECT = 0x100000;
+pub const O_LARGEFILE = 0;
+pub const O_NOATIME = 0x200000;
+pub const O_PATH = 0x1000000;
+pub const O_TMPFILE = 0x2010000;
+pub const O_NDELAY = O_NONBLOCK | 0x4;
+
+pub const F_DUPFD = 0;
+pub const F_GETFD = 1;
+pub const F_SETFD = 2;
+pub const F_GETFL = 3;
+pub const F_SETFL = 4;
+
+pub const F_SETOWN = 5;
+pub const F_GETOWN = 6;
+pub const F_GETLK = 7;
+pub const F_SETLK = 8;
+pub const F_SETLKW = 9;
+
+pub const F_RDLCK = 1;
+pub const F_WRLCK = 2;
+pub const F_UNLCK = 3;
+
+pub const F_SETOWN_EX = 15;
+pub const F_GETOWN_EX = 16;
+
+pub const F_GETOWNER_UIDS = 17;
+
+pub const LOCK_SH = 1;
+pub const LOCK_EX = 2;
+pub const LOCK_NB = 4;
+pub const LOCK_UN = 8;
+
+/// stack-like segment
+pub const MAP_GROWSDOWN = 0x0200;
+
+/// ETXTBSY
+pub const MAP_DENYWRITE = 0x0800;
+
+/// mark it as an executable
+pub const MAP_EXECUTABLE = 0x1000;
+
+/// pages are locked
+pub const MAP_LOCKED = 0x0100;
+
+/// don't check for reservations
+pub const MAP_NORESERVE = 0x0040;
+
+pub const VDSO_CGT_SYM = "__vdso_clock_gettime";
+pub const VDSO_CGT_VER = "LINUX_2.6";
+
+pub const Flock = extern struct {
+    l_type: i16,
+    l_whence: i16,
+    l_start: off_t,
+    l_len: off_t,
+    l_pid: pid_t,
+};
+
+pub const msghdr = extern struct {
+    msg_name: ?*sockaddr,
+    msg_namelen: socklen_t,
+    msg_iov: [*]iovec,
+    msg_iovlen: u64,
+    msg_control: ?*c_void,
+    msg_controllen: u64,
+    msg_flags: i32,
+};
+
+pub const msghdr_const = extern struct {
+    msg_name: ?*const sockaddr,
+    msg_namelen: socklen_t,
+    msg_iov: [*]iovec_const,
+    msg_iovlen: u64,
+    msg_control: ?*c_void,
+    msg_controllen: u64,
+    msg_flags: i32,
+};
+
+pub const off_t = i64;
+pub const ino_t = u64;
+pub const mode_t = u32;
+
+/// atime, mtime, and ctime have functions to return `timespec`,
+/// because although this is a POSIX API, the layout and names of
+/// the structs are inconsistent across operating systems, and
+/// in C, macros are used to hide the differences. Here we use
+/// methods to accomplish this.
+pub const libc_stat = extern struct {
+    dev: u64,
+    ino: ino_t,
+    mode: u32,
+    nlink: usize,
+
+    uid: u32,
+    gid: u32,
+    rdev: u64,
+    __pad0: u32,
+
+    size: off_t,
+    blksize: isize,
+    blocks: i64,
+
+    atim: timespec,
+    mtim: timespec,
+    ctim: timespec,
+    __unused: [2]isize,
+
+    pub fn atime(self: libc_stat) timespec {
+        return self.atim;
+    }
+
+    pub fn mtime(self: libc_stat) timespec {
+        return self.mtim;
+    }
+
+    pub fn ctime(self: libc_stat) timespec {
+        return self.ctim;
+    }
+};
+
+pub const kernel_stat = extern struct {
```

Comment:
This is wrong, check [the kernel definition](https://elixir.bootlin.com/linux/v5.3-rc8/source/arch/sparc/include/uapi/asm/stat.h#L26).

>  // Hack to make the stdlib not complain about atime

Arr, this is not needed, just use the correct structure definition.

Reply:
Sorry, stat vs stat64 distinction is a confusing one for me. 

> Arr, this is not needed, just use the correct structure definition.

What I meant is that the stdlib reads the timestamps using atime/ctime/mtime methods, not through direct field reads, so if those three methods aren't defined, the compiler will complain with something like:
```
lib/std/fs/file.zig:300:25: error: type 'u64' not a function
        const atime = st.atime();
                        ^
```
I guess I could make the stdlib reads the fields directly instead of going through methods like this, but the stat structs for other architectures need to be fixed first...

Reply:
Anyway, submitted a PR at #6986 to fix this.